proposals/0002-cxx-attributes.md

```diff
@@ -228,7 +228,125 @@ following grammar formulations are valid:
   \terminal{using} \textit{identifier} \opt{attribute-specifier-seq} \terminal{=} \textit{type-id} \terminal{;}\br
 \end{grammar}
 ```
-
 ![Latex Rendering](0002-assets/ClassGrammarRender.png)
 
+### Attribute Specification Language
+
+Attributes annotate source constructs with information. An attribute is said to
+be _applied to_ the entity or statement identified by the source construct.
+
+Some attributes may be required by an implementation for correct code
+generation, others may be optionally ignored. An implementation must issue a
+diagnostic on all ignored attributes unless otherwise specified by the
+definition of the attribute behavior.
+
+> Note: an example here would be optimization hint attributes which an
+> implementation is allowed to ignore without diagnosing.
+
+Each attribute may specify specific behavior for parsing attribute arguments.
```

Comment:
After reading the full PR, I'm still not really sure what this is trying to say. 

Are we saying that "special attributes may treat arguments specially"? Because if so, won't every attribution treat its arguments specially? If they weren't special, then the args would just be dropped?

Or is there some special attribute form I'm missing somewhere?

Reply:
Ah thanks, that helps me understand.

Just kind of winging it here:


Attributes that require arguments may also require custom parsing rules, such as the [availability](https://clang.llvm.org/docs/AttributeReference.html#availability) attribute. Attributes without arguments and attributes with trivial parsing such as integer literals do not need custom parsing and may rely on the general behavior described here.